src/Serializer.php

```diff
@@ -272,4 +277,24 @@ protected function getIncluded()
         $include = $this->request->get($this->expandParam);
         return is_string($include) ? preg_split('/\s*,\s*/', $include, -1, PREG_SPLIT_NO_EMPTY) : [];
     }
+
+    /**
+     * Checks Identification of resource object.
+     * http://jsonapi.org/format/#document-resource-object-identification
+     * @param ResourceInterface $resource
+     */
+    protected function checkIdentification(ResourceInterface $resource)
+    {
+        if (!is_string($resource->getId())) {
```

Comment:
>IMHO: ResourceModel shouldn't know about JSON-API - Trait - it's mixin
I agree about that. But it is not a responsibility of Serializer to return valid types of properties either. Also i don't like the idea to enforce developers to have only strings. Even if it is prohibited by specification.